modules/cert-manager-acme-dns01-aws.adoc

```diff
@@ -0,0 +1,187 @@
+
+:_content-type: PROCEDURE
+[id="cert-manager-acme-dns01-aws_{context}"]
+= Configuring an ACME issuer to solve DNS-01 challenges
+
+You can use {cert-manager-operator} to set up an ACME issuer to solve DNS-01 challenges. This procedure uses _Let's Encrypt_ as the ACME CA server and shows how to solve DNS-01 challenges with Amazon Route 53.
+
+.Prerequisites
+
+* You have access to the {product-title} cluster as a user with the `cluster-admin` role.
+* You have set up an IAM role for Amazon Route 53. For more information, see link:https://cert-manager.io/docs/configuration/acme/dns01/route53/[Route53] in the upstream cert-manager documentation.
++
+[NOTE]
+====
+For the {cert-manager-operator}, you must provide an `accessKeyID` and `secretAccessKey` to specify the credentials.
+====
+
+.Procedure
+
+. Configure the nameserver settings for the DNS-01 self check:
+
+.. Edit the `CertManager` resource by running the following command:
++
+[source,terminal]
+----
+$ oc edit certmanager cluster
+----
+
+.. Add a `spec.controllerConfig` section with the following override arguments:
++
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1alpha1
+kind: CertManager
+metadata:
+  name: cluster
+  ...
+spec:
+  ...
+  controllerConfig:                                <1>
+    overrideArgs:
+      - '--dns01-recursive-nameservers=1.1.1.1:53' <2>
+      - '--dns01-recursive-nameservers-only'       <3>
+----
+<1> Add the `spec.controllerConfig` section.
+<2> Provide a comma-separated list of `<host>:<port>` nameservers to query for the DNS-01 self check.
+<3> Specify to only use recursive nameservers instead of checking the authoritative nameservers associated with that domain.
++
+// TODO: Upstream docs say that this might take longer using this - determine if that's true/state here if so in a future update
+
+.. Save the file to apply the changes.
+
+. Create a namespace for the issuer.
+
+.. Create a YAML file that defines a `Namespace` object:
++
+.Example `namespace.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: my-issuer-namespace <1>
+----
+<1> Specify the namespace for the issuer.
+
+.. Create the `Namespace` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f namespace.yaml
+----
+
+. Create a secret to store your AWS credentials in.
+
+.. Create a YAML file that defines a `Secret` object:
++
+.Example `secret.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: "aws-secret"                          <1>
+  namespace: my-issuer-namespace              <2>
+type: Opaque
+data:
+  awsSecretAccessKey: <aws_secret_access_key> <3>
+----
+<1> Specify the name for the secret.
+<2> Specify the namespace for the secret.
+<3> Replace `<aws_secret_access_key>` with your AWS secret access key.
+
+.. Create the `Secret` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f secret.yaml
+----
```

Comment:
> <3> Replace `<aws_secret_access_key>` with your AWS secret access key.

This statement is incorrect. Within a "Secret" resource, the field's value will be the base64-encoded value of the access key. You will have to document the base64 command if needed for readers. For simplicity, a command is better than above many lines:
```suggestion
[source,terminal]
----
$ oc create secret generic aws-secret --from-literal=awsSecretAccessKey=<aws_secret_access_key> -n my-issuer-namespace
----
```

Reply:
@xingxingxia Thanks, I updated to use the command. With using the command, if I use that callout now, would that be correct? (I'm pushing an update that will use it, but let me know if I need to say something different).